node/runtime/src/lib.rs

```diff
@@ -872,6 +882,177 @@ impl pallet_contracts::Config for Runtime {
+		1.into()
+	}
+}
+
+parameter_types! {
+	pub const EthChainId: u64 = 7;
+}
+
+/// Clone of Istanbul config with `create_contract_limit` raised.
+static EVM_CONFIG: EvmConfig = EvmConfig {
+	gas_ext_code: 700,
+	gas_ext_code_hash: 700,
+	gas_balance: 700,
+	gas_sload: 800,
+	gas_sstore_set: 20000,
+	gas_sstore_reset: 5000,
+	refund_sstore_clears: 15000,
+	gas_suicide: 5000,
+	gas_suicide_new_account: 25000,
+	gas_call: 700,
+	gas_expbyte: 50,
+	gas_transaction_create: 53000,
+	gas_transaction_call: 21000,
+	gas_transaction_zero_data: 4,
+	gas_transaction_non_zero_data: 16,
+	sstore_gas_metering: true,
+	sstore_revert_under_stipend: true,
+	err_on_call_with_more_gas: false,
+	empty_considered_exists: false,
+	create_increase_nonce: true,
+	call_l64_after_gas: true,
+	stack_limit: 1024,
+	memory_limit: usize::max_value(),
+	call_stack_limit: 1024,
+	create_contract_limit: None,
+	call_stipend: 2300,
+	has_delegate_call: true,
+	has_create2: true,
+	has_revert: true,
+	has_return_data: true,
+	has_bitwise_shifting: true,
+	has_chain_id: true,
+	has_self_balance: true,
+	has_ext_code_hash: true,
+	estimate: false,
+};
+
+impl pallet_evm::Config for Runtime {
+	type FeeCalculator = FixedGasPrice;
+	type CallOrigin = EnsureAddressTruncated;
+	type WithdrawOrigin = EnsureAddressTruncated;
+	type AddressMapping = HashedAddressMapping<BlakeTwo256>;
+	type Currency = Balances;
+	type Event = Event;
+	type Runner = pallet_evm::runner::stack::Runner<Self>;
+	type Precompiles = EdgewarePrecompiles;
+	type ChainId = EthChainId;
+	type GasToWeight = ();
+
+	/// EVM config used in the module.
+	fn config() -> &'static EvmConfig {
+		&EVM_CONFIG
+	}
+}
+
+pub struct EdgewarePrecompiles;
+
+type PrecompiledCallable = fn(&[u8], Option<usize>)
+	-> core::result::Result<(pallet_evm::ExitSucceed, Vec<u8>, usize), pallet_evm::ExitError>;
+
+impl pallet_evm::Precompiles for EdgewarePrecompiles {
+	fn execute(
+		address: H160,
+		input: &[u8],
+		target_gas: Option<usize>
+	) -> Option<core::result::Result<(pallet_evm::ExitSucceed, Vec<u8>, usize), pallet_evm::ExitError>> {
+		match get_precompiled_func_from_address(&address) {
+			 Some(func) => return Some(func(input, target_gas)),
+			 _ => {},
+		};
+
+		None
+	}
+}
+
+fn get_precompiled_func_from_address(address: &H160) -> Option<PrecompiledCallable> {
+	use core::str::FromStr;
+	use pallet_evm::Precompile;
+
+	// identity
+	let addr_ecrecover = H160::from_str("0000000000000000000000000000000000000001").expect("Invalid address at precompiles generation");
+	// ecrecover
+	let addr_sha256 = H160::from_str("0000000000000000000000000000000000000002").expect("Invalid address at precompiles generation");
+	// sha256
+	let addr_ripemd160 = H160::from_str("0000000000000000000000000000000000000003").expect("Invalid address at precompiles generation");
+	// ripemd160
+	let _addr_modexp = H160::from_str("0000000000000000000000000000000000000004").expect("Invalid address at precompiles generation");
```

Comment:
can we remove all these precompiles under ripemd160 for now. The PR never got merged into Substrate and don't exist yet.